app/components/CopySelectionTooltip.tsx

```diff
@@ -0,0 +1,141 @@
+      if (!sel || sel.rangeCount === 0) return false;
+      const range = sel.getRangeAt(0);
+      const commonAncestor = range.commonAncestorContainer as Node;
+      return root.contains(
+        commonAncestor.nodeType === Node.ELEMENT_NODE
+          ? (commonAncestor as Element)
+          : (commonAncestor.parentElement as Element)
+      );
+    },
+    []
+  );
+
+  const updateFromSelection = useCallback(() => {
+    if (!enabled) return;
+    const root = containerRef.current;
+    if (!root) return;
+    const sel = window.getSelection();
+    if (!sel || sel.rangeCount === 0 || sel.isCollapsed) {
+      setVisible(false);
+      return;
+    }
+    // Ensure selection is inside our container and not inside inputs/textareas
+    if (!isSelectionInside(root, sel)) {
+      setVisible(false);
+      return;
+    }
+    const range = sel.getRangeAt(0);
+    const rect = range.getBoundingClientRect();
+    if (!rect || (rect.width === 0 && rect.height === 0)) {
+      setVisible(false);
+      return;
+    }
+    const rootRect = root.getBoundingClientRect();
+    const tooltipHeight = tooltipRef.current?.offsetHeight || 0;
+    const gap = 8; // space between selection and tooltip
+    const top =
+      rect.top - rootRect.top - (tooltipHeight > 0 ? tooltipHeight + gap : 48);
+    const left = Math.max(0, rect.left - rootRect.left);
+    setPosition({ top, left });
+    setVisible(true);
+  }, [enabled, isSelectionInside]);
+
+  useEffect(() => {
+    const onScroll = () => setVisible(false);
+    document.addEventListener("selectionchange", updateFromSelection);
+    window.addEventListener("scroll", onScroll, true);
+    window.addEventListener("resize", onScroll);
+    return () => {
+      document.removeEventListener("selectionchange", updateFromSelection);
+      window.removeEventListener("scroll", onScroll, true);
+      window.removeEventListener("resize", onScroll);
+    };
+  }, [updateFromSelection]);
+
+  const handleCopy = useCallback(async () => {
+    const sel = window.getSelection();
+    const text = sel?.toString() ?? "";
+    if (!text) return;
+    try {
+      await navigator.clipboard.writeText(text);
+      // Hide after copy
+      setVisible(false);
+    } catch {
+      // noop: copying may be blocked; we simply leave the tooltip
+    }
+  }, []);
+
+  return (
+    <Box position="relative" ref={containerRef}>
+      {children}
+      {enabled && visible && (
+        <Box
+          position="absolute"
+          top={`${position.top}px`}
+          left={`${position.left}px`}
+          zIndex="tooltip"
+          bg="bg"
+          color="fg"
+          borderRadius="md"
+          shadow="md"
+          borderWidth="1px"
+          borderColor="border"
+          maxW={{ base: "calc(100% - 16px)", md: "560px" }}
+          p="2.5"
+          ref={tooltipRef}
+        >
+          <Flex align="center" gap="2">
+            <Icon as={SparkleIcon} color="fg.muted" />
```

Comment:
```suggestion
            <Icon as={SparkleIcon} color="secondary.500" />
```
@kamicut yes, secondary.500